src/tests/system/tests/test_ldap.py

```diff
@@ -454,3 +455,55 @@ def test_ldap__lookup_and_authenticate_as_user_with_different_object_search_base
     assert result is not None, "User is not found!"
     assert result.name == user.name, "Username is not correct!"
     assert client.auth.ssh.password(user.name, "Secret123"), "User login failed!"
+
+@pytest.mark.ticket(jira="RHEL-55993")
+@pytest.mark.importance("critical")
+@pytest.mark.parametrize("modify_mode", ["exop", "exop_force"])
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_ldap__password_change_no_grace_logins_left(
+    client: Client, ldap: LDAP, modify_mode: str
+):
+    """
+    :title: Password change when no grace logins left
+    :setup:
+        1. Set "passwordExp" to "on"
+        2. Set "passwordMaxAge" to "1"
+        3. Set "passwordGraceLimit" to "0"
+        4. Add a user to LDAP
+        5. Wait until the password is expired
+        6. Set "ldap_pwmodify_mode"
+        7. Start SSSD
+    :steps:
+        1. Authenticate as the user with 'exop_force' set
+        1. Authenticate as the user with 'exop' set
+    :expectedresults:
+        1. With 'exop_force' expect a password change which will fail
+        2. With 'exop' expect just a failed login
+    :customerscenario: False
+    """
+    ldap.ldap.modify("cn=config", replace={"passwordExp": "on"})
+    ldap.ldap.modify("cn=config", replace={"passwordMaxAge": "1"})
+    ldap.ldap.modify("cn=config", replace={"passwordGraceLimit": "0"})
+    ldap.user("user1").add(password="Secret123")
+
+    # make sure the password is expired
+    time.sleep(3)
+
+    client.sssd.domain["ldap_pwmodify_mode"] = modify_mode
+    client.sssd.start()
+
+    match modify_mode:
+        case "exop_force":
+            raised = False
+            try:
+                assert not client.auth.ssh.password_expired(
+                    "user1", "Secret123", "red_32"
+                ), "Password should not have been able to be changed!"
+            except ExpectScriptError as e:
+                assert e.code == 202, "Unexpected error code"
+                raised = True
+            assert raised, "No expection raised"
```

Comment:
FYI this could be written as
```
with pytest.raises(ExpectScriptError) as e:
    assert e.code == 202, "Unexpected error code"
```

Reply:
Hi,

thanks for the hint about needed change in the test framework. Since it is important for the test to see the reason for the failure I added the changes to `password_with_output()` method since this can return actual error code and not only true/false. I created https://github.com/SSSD/sssd-test-framework/pull/129 with the change.

bye,
Sumit

Reply:
Ok, IIUC you want to call password_with_output() and check that password change was requested?

Reply:
Hi,

yes, for `exop` the authentication will just fail, i.e. return code is `1`, and for `exop_force` the passowrd change dialog should start. Since I think that the password change dialog is not expected when calling `password_with_output()` I treated this as error with the new return code `3`.

bye,
Sumit

Reply:
Got it. I acked and pushed the framework change.